### ISSUE TITLE
Problem: mls functionality not checked against reference implementation

### ISSUE THREAD

**Issue by tomtau** [crypto-com/chain#1848]
if possible, messages from this test https://github.com/crypto-com/chain/pull/1795/files#diff-a663bb86ce2220f56e2d0a1ac4adb0beR981
could potentially be tried with the reference implementation whether the same state is derived

**Comment by tomtau** [comment]
X509 credential may not be supported in reference implementation

**Comment by tomtau** [comment]
there are also a bunch of recent changes in the draft:
- key schedule was reordered
- keypackage must contain `capabilities` extension
- the tree isn't by default provided in Welcome -- it can be optionally provided in an extension of type `ratchet_tree`
- ...

**Comment by yihuang** [comment]
> there are also a bunch of recent changes in the draft:
> 
> * key schedule was reordered
> * keypackage must contain `capabilities` extension
> * the tree isn't by default provided in Welcome -- it can be optionally provided in an extension of type `ratchet_tree`
> * ...

I think go-mls not up to date with spec, I guess it's more time efficient for us to catch up with spec first?

**Comment by tomtau** [comment]
yeah, opened this: https://github.com/crypto-com/chain/issues/2040
